### PR TITLE
Issue 86 #2 add an option to augment developer written tests

### PR DESCRIPTION
**Pull request by haim-kermany**
## Description

we merge the user test suite exec file with the ctd test exec file.
the coverage of the merged exec file is used as the coverage of the base tests the we augment to
Related to # 86

## Type of Change

Please check the types of changes your PR introduces.

- [ ] Bug fix (non-breaking change that fixes an issue)
- [X ] New feature (non-breaking change that adds functionality)
- [ ] Refactoring (non-breaking code restructuring that preserves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build-related update (CI workflow, test cases)
- [ ] Documentation update
- [ ] Other (please describe):

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so that it can be replicated.
Please also list any relevant details for your test configuration.

added a test:
test_generate_ctdamplified_evosuite_allclasses_augmentcoverage_with_user_coverage

- [ ] Test A
- [ ] Test B

## Checklist

- [X ] My code follows the style guidelines of this project
- [X ] I have performed a self-review of my own code
- [X ] I have commented my code, particularly in hard-to-understand areas
- [X ] I have made corresponding changes to the documentation
- [X ] I have added tests that prove my fix is effective or that my feature works
- [X ] New and existing tests pass locally with my changes
- [X ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tkltest/util/coverage_util.py [hunk 1]
```diff
@@ -44,28 +46,88 @@ def get_coverage_for_test_suite(build_file, build_type, test_root_dir, report_di
...
+        else:  # gradle
+            cmd = "gradle --project-dir {} {}".format(os.path.dirname(additional_build_file), additional_build_targets)
+        try:
+            command_util.run_command(cmd, verbose=False)
+        except subprocess.CalledProcessError as e:
+            tkltest_status('Error while running test suite for coverage computing:\n {}\n{}'.format(e, e.stderr))
+            no_failure = False
+        additional_exec_file = additional_test_suite['coverage_exec_file']
+        if no_failure and not os.path.exists(additional_exec_file):
+            tkltest_status('Error: {} was not created using command: {}'.format(additional_exec_file, cmd))
```
Comment (sinha108) [inline tkltest/util/coverage_util.py:104]: @haim-kermany: For printing error messages, the preferred way is to set the option `error=True` in the call to `tkltest_status`. This option causes `ERROR: `to be prepended to the message; the message parameter need not contain the `Error:` prefix. See https://github.com/konveyor/tackle-test-generator-cli/blob/main/tkltest/util/logging_util.py#L36-L40

**Review by sinha108 — CHANGES_REQUESTED** [review]
For consistency, the printing of error messages should call `tkltest_status` with option `error=True`

**Review by rachelt44 — CHANGES_REQUESTED** [review]
Fail to get coverage information from additional_test_suite {}, using only coverage from ->
Failed to obtain coverage from dev-written test suite using {}, using only coverage from ctd-amplified test suite {}

**Review by rachelt44 — CHANGES_REQUESTED** [review]
fail to merge {} with {}: -> Failed to merge coverage data files {} and {]:
fail to obtain {} from {} -> Failed to create CSV coverage report {} from {}

**Review by rachelt44 — APPROVED** [review]

**Review by rachelt44 — CHANGES_REQUESTED** [review]
if os.path.exists(additional_exec_file): ->
if not os.path.exists(additional_exec_file):

Also don't flag it as an error, add Warning to the message, it's only a warning because we continue running

**Review by rachelt44 — APPROVED** [review]

**Review by sinha108 — APPROVED** [review]

**Pull request merged**